core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -396,7 +399,9 @@ public void commit() {
     } catch (CommitStateUnknownException commitStateUnknownException) {
       throw commitStateUnknownException;
     } catch (RuntimeException e) {
-      Exceptions.suppressAndThrow(e, this::cleanAll);
+      if (!strictCleanup || e instanceof CleanableFailure) {
+        Exceptions.suppressAndThrow(e, this::cleanAll);
```

Comment:
just curious, would we need check for `strictCleanup` in https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/BaseTransaction.java#L453-L463 as well?

Reply:
Yes! Good catch. Amogh is fixing this in the updated PR.